tests/Integration/CalculatingCartTotalTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+namespace Tests\Integration;
+use App\Models\Product;
+use App\Repositories\ProductRepository;
+use App\Repositories\OfferRepository;
+use App\Repositories\CartRepository;
+use App\Repositories\CartItemRepository;
+use App\Models\CartItem;
+use App\Models\Cart;
+use App\Services\OrderService;
+
+class calculateCartTotalTest extends \PHPUnit\Framework\TestCase
+{
+    public function testCalculateTotalReturnsCorrectTotal()
+    {
+        $databaseHelper = new \App\Core\Database();
+        
+        $productRepository = new ProductRepository($databaseHelper);
+        $offerRepository = new OfferRepository($databaseHelper);
+        $cartItemRepository = new CartItemRepository($databaseHelper);
+        $cartRepostiry = new CartRepository($databaseHelper);
+
+        $product1 = new Product();
+        $product1->setName('Product 1');
+        $product1->setPrice(4000);
+        $product1->setCode('P1');
+
+        $product2 = new Product();
+        $product2->setName('Product 2');
+        $product2->setPrice(6000);
+        $product2->setCode('P2');
```

Comment:
We could move these into a private method with the product data passed to the method